### PR TITLE
HW 7: Task 3

### PR DESCRIPTION
**Pull request by ValeriL**

### CODE REVIEW

File: Homework/homework7/task3.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    if amount_winners == 1:
+        return winners[0] + " wins!"
+    if amount_winners == 2:
+        return "draw"
+
+    return "unfinished"
+
+
+def tic_tac_toe_checker(board: List[List]) -> str:
+    lined_board = sum(board, [])
```
Comment (Cjkjvfnby) [inline Homework/homework7/task3.py:34]: Usually, the prefix `flat` is used in that case.

File: Homework/homework7/task3.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        (0, 3, 6),
+        (1, 4, 7),
+        (2, 5, 8),
+    ]
+    winners = []
+
+    for comb in win_idx_comb:
+        cur_comb = [lined_board[idx] for idx in comb]
+        cur_sign = cur_comb[0]
+        if len(set(cur_comb)) == 1 and cur_sign != "-":
```
Comment (Cjkjvfnby) [inline Homework/homework7/task3.py:50]: Can you avoid steps when you need to convert the list to set?

File: Homework/homework7/task3.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    Return value should be "unfinished"
+    [[-, -, o],
+     [-, o, o],
+     [x, x, x]]
+     Return value should be "x wins!"
+"""
+from typing import List
+
+
+def result(winners: List) -> str:
```
Comment (Cjkjvfnby) [inline Homework/homework7/task3.py:22]: can we get a more descriptive name for a function?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: Homework/homework7/task3.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        (0, 3, 6),
+        (1, 4, 7),
+        (2, 5, 8),
+    ]
+    winners = []
+
+    for comb in win_idx_comb:
+        cur_comb = [flat_board[idx] for idx in comb]
+        cur_sign = cur_comb[0]
+        if cur_comb.count(cur_sign) == 3 and cur_sign != "-":
```
Comment (Cjkjvfnby) [inline Homework/homework7/task3.py:50]: Do you remember about set comprehension?

**Review by Cjkjvfnby — APPROVED** [review]

File: Homework/homework7/task3.py (hunk 4 above)
Comment (ValeriL) [reply Homework/homework7/task3.py:50]: Ah, got it, thank you

**Pull request merged**